activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/conf/TaskRuntimeAutoConfiguration.java

```diff
@@ -129,6 +128,16 @@ public InitializingBean registerTaskCreatedEventListener(RuntimeService runtimeS
                                                                                      taskCreatedEventConverter),
                                                      ActivitiEventType.TASK_CREATED);
     }
+    
+    @Bean
+    public InitializingBean registerTaskUpdatedEventListener(RuntimeService runtimeService,
+                                                             @Autowired(required = false) List<TaskRuntimeEventListener<TaskUpdatedEvent>> listeners,
+                                                             ToAPITaskUpdatedEventConverter taskCreatedEventConverter) {
+        return () -> runtimeService.addEventListener(new TaskUpdatedListenerDelegate(getInitializedTaskRuntimeEventListeners(listeners),
+                                                                                     taskCreatedEventConverter),
+                                                     ActivitiEventType.ENTITY_UPDATED);
```

Comment:
@igdianov @salaboy if I remember well we moved away from this solution (listening to internal engine events), because for a task, when it has an assignee in the definition, the event 'ENTITY_UPDATED' happens before the event `TASK_CREATED` event.
We have: `ENTITY_CREATED`, `ENTITY_UPDATED`, `TASK_CREATED`). This will make the current implementation of query-service fail. If we want keep listening to internal events we need either:
- use to `ENTITY_CREATED` (instead of TASK_CREATED) to trigger `TaskCreatedEvent`. In this case we need to check that `ENTITY_CREATED` event provides exactly the same info as `TASK_CREATED`. We may need to re-trigger an event on `TASK_CREATED` to update info if necessary.

**Or**

- handle taskUpdatedEvent as a taskCreatedEvent in the case where no task is found in query-service.

**Or**

- any other solution?

The first option (`ENTITY_CREATED`) seems be the better approach IMO.

Reply:
I fixed it in TaskEntityManagerImpl.java, that was ignoring fireEvents flag

Reply:
Awesome @igdianov I've missed this small change in `TaskEntityManagerImpl` that fixed the ordering issue!